flash/agent/agent.py

```diff
@@ -50,14 +69,32 @@ def write_section(self, section: Prompt, technology_name: str) -> str:
         return chat_response.content
     
     def generate_section(self, section_title: Prompt) -> str:
+        """
+        Generates a section of the learning material.
+
+        Parameters:
+            section_title (Prompt): The title prompt of the section.
+
+        Returns:
+            str: The content of the generated section.
+        """
         logging.info(f"Writing {section_title.value} section")
 
         section_content = self.write_section(section_title, self.technology)
         self.learning_material.set(section_title, section_content)
 
         return section_content
     
-    def generate_full(self) -> str:
+    def generate_full(self) -> None:
+        """
+        Generates the full learning material.
+
+        Parameters:
+            None
+
+        Returns:
+            None
```

Comment:
As you can see from `LearningMaterial.render()`, it returns a string. Please update this docstring accordingly.